lib/datastore/dataset.js

```diff
@@ -168,23 +168,60 @@ Dataset.prototype.determineApiEndpoint_ = function(customApiEndpoint) {
  * @return {Key} A newly created Key from the options given.
  *
  * @example
- * var key;
- *
- * // Create an incomplete key from the dataset namespace, kind='Company'
- * key = dataset.key('Company');
+ * //-
+ * // Create an incomplete key with a kind value of `Company`.
+ * //-
+ * var key = dataset.key('Company');
  *
- * // A complete key from the dataset namespace, kind='Company', id=123
- * key = dataset.key(['Company', 123]);
+ * //-
+ * // Create a complete key with a kind value of `Company` and id `123`.
+ * //-
+ * var key = dataset.key(['Company', 123]);
  *
- * // A complete key from the dataset namespace, kind='Company', name='Google'
- * // Note: `id` is used for numeric identifiers and `name` is used otherwise
- * key = dataset.key(['Company', 'Google']);
+ * //-
+ * // Create a complete key with a kind value of `Company` and name `Google`.
+ * // Note: `id` is used for numeric identifiers and `name` is used otherwise.
+ * //-
+ * var key = dataset.key(['Company', 'Google']);
  *
- * // A complete key from a provided namespace and path.
- * key = dataset.key({
+ * //-
+ * // Create a complete key from a provided namespace and path.
+ * //-
+ * var key = dataset.key({
  *   namespace: 'My-NS',
  *   path: ['Company', 123]
  * });
+ *
+ * //-
+ * // All keys have helper properties that you can access or overwrite
+ * // directly.
+ * //-
+ * var key = dataset.key(['Company', 123]);
+ *
+ * // key.path: ['Company', 123]
```

Comment:
Instead of trying to document all of these inline, would it make more sense to make a Key section within the docs, then we can document them JSDoc style?